02-esc/challenges.js

```diff
@@ -197,6 +197,7 @@ function showAllChallenges(resultFromAPI) {
 		btnBook.setAttribute("data-id", id);
 		btnBook.setAttribute("data-minparticipants", minParticipants);
 		btnBook.setAttribute("data-maxparticipants", maxParticipants);
+		btnBook.setAttribute("data-title", title);
```

Comment:
Storing data like this in the DOM is unconventional, and I'm not sure what to think of it. On one hand, it blurs the lines between data and logic on one hand, and presentation on the other hand. At the same time, using `data-` attributes clearly marks the DOM attributes as containing data that is used by code, as opposed to just grabbing the data from an ID attribute or something like that.

Regardless of how to think about data in DOM _on principle_, storing data about the challenge on a button feels weird, because the data is not related to the button itself, but to the challenge as a whole.